src/promptflow/promptflow/azure/operations/_run_operations.py

```diff
@@ -682,20 +687,18 @@ def _resolve_automatic_runtime(self, run, session_id, reset=None):
             "it may take a while to build runtime and request may fail with timeout error. "
             "Wait a while and resubmit same flow can successfully start the run."
         )
-        runtime_name = "automatic"
+        runtime_name = AUTOMATIC_RUNTIME_NAME
         self._resolve_session(run=run, session_id=session_id, reset=reset)
         return runtime_name
 
     def _resolve_runtime(self, run, flow_path, runtime, reset=None):
         runtime = run._runtime or runtime
         session_id = self._get_session_id(flow=flow_path)
 
-        if runtime:
-            if not isinstance(runtime, str):
-                raise TypeError(f"runtime should be a string, got {type(runtime)} for {runtime}")
-        else:
+        if runtime is None or runtime == AUTOMATIC_RUNTIME_NAME:
             runtime = self._resolve_automatic_runtime(run=run, session_id=session_id, reset=reset)
-
+        elif not isinstance(runtime, str):
+            raise TypeError(f"runtime should be a string, got {type(runtime)} for {runtime}")
```

Comment:
If no error raised, maybe worth a log to print what runtime is specified? Ignore this if it's printed in other place.

Reply:
Offline synced, the runtime info already exists in flow run output dict.